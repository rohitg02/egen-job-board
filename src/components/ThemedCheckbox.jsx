import React from 'react';
import PropTypes from 'prop-types';

import { Checkbox, Typography } from '@material-ui/core';

const ThemedCheckbox = ({onChangeHandler, label, value}) => {
  return (
    <React.Fragment>
      <Checkbox
        if="themedCheckbox"
        checked={value}
        color="primary"
        onChange={() => onChangeHandler(!value)}
        inputProps={{ 'aria-label': 'primary checkbox' }}
      />
      <Typography noWrap>
          {label}
      </Typography>
    </React.Fragment>
  );
}

ThemedCheckbox.propTypes ={
  onChangeHandler: PropTypes.func,
  label: PropTypes.string,
  value: PropTypes.bool
}

ThemedCheckbox.defaultProps ={
  onChangeHandler: () => {},
  label: '',
  value: false
}

export default ThemedCheckbox;
