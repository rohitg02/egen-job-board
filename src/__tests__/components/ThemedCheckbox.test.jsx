import React from 'react';
import { shallow } from '../../enzyme';

import ThemedCheckbox from '../../components/ThemedCheckbox';

describe('ThemedCheckbox component', () => {

  it('should render ThemedCheckbox component', () => {
    const wrapper = shallow(<ThemedCheckbox />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default onChangeHandler', () => {
    expect(ThemedCheckbox.defaultProps.onChangeHandler).toBeDefined();
  });

});
