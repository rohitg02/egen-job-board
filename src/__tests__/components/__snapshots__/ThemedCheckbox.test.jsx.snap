// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemedCheckbox component should render ThemedCheckbox component 1`] = `ShallowWrapper {}`;
